src/Arcus.WebApi.OpenApi.Extensions/OAuthAuthorizeOperationFilter.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.AspNetCore.Authorization;
+using Swashbuckle.AspNetCore.Swagger;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace Arcus.WebApi.OpenApi.Extensions
+{
+    /// <summary>
+    /// A Swashbuckle operation filter that adds OAuth security definitions to authorized API operations.
+    /// </summary>
+    public class OAuthAuthorizeOperationFilter : IOperationFilter
+    {
+        private readonly IEnumerable<string> _scopes;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OAuthAuthorizeOperationFilter"/> class.
+        /// </summary>
+        public OAuthAuthorizeOperationFilter(IEnumerable<string> scopes)
+        {
+            _scopes = scopes;
+        }
+
+        public void Apply(Operation operation, OperationFilterContext context)
+        {
+            var hasAuthorize = context.MethodInfo.GetCustomAttributes(true).OfType<AuthorizeAttribute>().Any() ||
+                               (
+                                   context.MethodInfo.DeclaringType.GetCustomAttributes(true).OfType<AuthorizeAttribute>().Any() &&
+                                   context.MethodInfo.GetCustomAttributes(false).OfType<AllowAnonymousAttribute>().Any() == false
+                               );
+
+            if (hasAuthorize)
+            {
+                if (operation.Responses.ContainsKey("401") == false)
```

Comment:
This is interesting, and maybe something for more down the road, but do we add both or make it configurable?

Reply:
You mean adding the 401 & 403 possible response types ?
I'd add them both for now and we could have a look at making it configurable when there's a real requirement or feature request to make it configurable :)

Reply:
Yes, I agree!